src/main/resources/test_files/runtime-spark/DataFrames/ForClause2.jq

```diff
@@ -1,2 +1,2 @@
 (:JIQS: ShouldRun; Output="({ "$i" : 1, "$j" : 4 }, { "$i" : 1, "$j" : 5 }, { "$i" : 1, "$j" : 6 }, { "$i" : 2, "$j" : 4 }, { "$i" : 2, "$j" : 5 }, { "$i" : 2, "$j" : 6 }, { "$i" : 3, "$j" : 4 }, { "$i" : 3, "$j" : 5 }, { "$i" : 3, "$j" : 6 })" :)
-for $i in parallelize((1,2,3)) for $j in (4,5,6) return {"$i": $i, "$j": $j}
+for $i in parallelize((1,2,3)) for $j in (4,5,6) order by $i, $j return {"$i": $i, "$j": $j}
```

Comment:
The change of order may be triggered by the switch to cartesian product -- since normally the order is supposed to be deterministic, this actually indicates that we should not use the cartesian product if the new for clause is local, but stick to the old code. That is, cartesian product only if (i) child is a DF, (ii) for expression does not depend on variables bound in current FLWOR and (iii) for expression is an RDD. For this case, we can live with indeterministic outputs. Does it make sense?

Reply:
Yes, this seems very sound now